docs/advanced-features/i18n-routing.md

```diff
@@ -214,6 +214,12 @@ export default function IndexPage(props) {
 }
 ```
 
+## Leveraging the NEXT_LOCALE cookie
+
+Next.js supports overriding the accept-language header with a `NEXT_LOCALE=the-locale` cookie. This cookie can be set using a language switcher and then when a user comes back to the site it will leverage the locale specified in the cookie.
```

Comment:
There is actually no note that this feature works only when user requesting `/`. That's why there are many issues about `NEXT_LOCALE` not working.

Reply:
Hi, I opened a PR here explaining this behavior more https://github.com/vercel/next.js/pull/22736